### ISSUE TITLE
Liquids in head don't slow you down

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#5841]
Long story short: If you only walk with your *head* through a liquid, you are not slowed down. It seems its only the feet which matter here.

Long story long: It is possible in some strange builds that you can have a non-walkable non-liquid node below a liquid nodes. Example in Minetest Game: Water Source on top of torch (you can even make a full walkway out of this). If you walk through it, your feet are in a non-liquid node while your head is in the liquid.

So far, so good. But it turns out, walking through this unusual setup of nodes does not slow you down like one might expect.

ef152428ef6ac609d805c4519b2247556db44931

**Comment by paramat** [comment]
I guess just 1 node is checked to keep it simple. Since the inverted situation is rare i think it's not worth adding a 2nd node check.

**Comment by nerzhul** [comment]
Yeah @Wuzzy2 i agree i should be done for feet not head, maybe we can look at this on next release

**Comment by dsohler** [comment]
It would be awesome to check for two nodes and using the average so it's either 0 (no node the player is in is water), or 1 (both nodes the player is in is water), or 0.5 (one of the nodes the player is in is water) as modifier for walking speed.

**Comment by HybridDog** [comment]
dsohler, l think this would be unrealistic because you'd just get 3 discrete slowing factors instead of 2. l'd prefer continual slowing.

**Comment by paramat** [comment]
> It is possible in some strange builds that you can have a non-walkable non-liquid node below a liquid nodes. Example in Minetest Game: Water Source on top of torch

It's a very rare situation though, and torches are now dropped by liquids in MTGame.
I don't think this is worth doing for extremely rare situations :-1:

**Comment by HybridDog** [comment]
That liquids don't flow through fences but stop next to them is more severe l think.

**Comment by paramat** [comment]
That's very difficult to address, and not worth trying, like liquids flowing through open doors or liquids flowing over plants. The door, fence and plants are nodes so no other node can occupy the same space.

**Comment by HybridDog** [comment]
Yes, it'd also change gameplay. If you slow players down when liquid is in their head and liquids flow through fences and doors, other "problems" would become drastically more salient, e.g. missing diagonal sunlight and shadows.
